### ISSUE TITLE
Indexing slows down with large numbers, e.g. unixtime numbers.

### ISSUE THREAD

**Issue by hastebrot** [oramasearch/orama#428]
I have 2000 documents with text that each are not larger than 65536 bytes. When I add just a single field with a number, e.g. using `createdAt: new Date(doc.createdAt).getTime(),` it slows down `insert()` considerably. interestingly, low numbers are fine.

indexing of these documents with number: 14 seconds, without numbers: 3 seconds.

**Comment by hastebrot** [comment]
As a work-around I can use `string`s instead with `createdAt: new Date(doc.createdAt).toISOString(),`. 

when I do a search and sort the ISO dates it will return the same result as with unixtime:

```
  const result3 = await search(docs, {
    sortBy: {
      property: "createdAt",
      order: "DESC",
    },
    limit: 100,
  });
```

**Comment by micheleriva** [comment]
I believe I am the one to blame for this. I opted for implementing [AVL trees](https://oramasearch.com/blog/avl-trees-explained) to improve performance at search time, but slowing down index time too much.

I will take some time to test better alternatives (maybe red-black trees?) to see if we can also improve indexing time.

**Comment by hastebrot** [comment]
A practical, performance, and easy-to-implement alternative to AVL and RB tree are Zip Trees [1] by Robert Tarjan. Maybe they are useful.

[1] https://arxiv.org/abs/1806.06726

**Comment by micheleriva** [comment]
Thanks a lot, @hastebrot! I'll look into this next week.

**Comment by micheleriva** [comment]
Started working on it: https://github.com/oramasearch/orama/pull/430

**Comment by rocwang** [comment]
I see the same issue if one indexed document field is typed as `number`.

I have about 80k documents. Each document is pretty small and has a `popularity` field indicating a popularity score.

* If I leave it as `number`, the indexing time via `insertMultiple` would be about 25s. 
* If I convert it to `string` padded with `0` on the left, the indexing time would just be about 3 to 4 seconds.

As per my experiment, it looks like the issue is not related to how big the number is though.

As I intended to do indexing in browser, this issue was almost a deal-breaker for me until I found the workaround in this page.
I really wish #430 would fix it.
